create_data_image.sh

```diff
@@ -30,27 +19,27 @@ echo
 # larger so the image stays small.
 IMAGE_SIZE=$((8 + ${TOTAL_SIZE} / (920*1024)))
 
+set -e
```

Comment:
Seeing `set -e` halfway the file is a bit weird. Wouldn't it make more sense to apply it to the entire script?

In general, I'm not a huge fan of `set -e`, because when it exits, the shell doesn't actually tell you where it aborted execution, or that it aborted execution at all. So it mostly just adds confusion when debugging. Are there things that can go terribly wrong by not aborting the script?

Reply:
> Seeing set -e halfway the file is a bit weird. Wouldn't it make more sense to apply it to the entire script?

Agreed and fixed. I was trying to keep changes to a minimum but in the end I've rewritten this whole file.

> In general, I'm not a huge fan of set -e

Not using `-e` (or `-euo pipefail` in Bash) in a pipeline script is like wrapping every Python statement in a try-catch that ignores the error. An exponential number of things can go wrong making it impossible to reason about.

It is a best practice to always use `set -euo pipefail` by default.

> So it mostly just adds confusion when debugging.

See this guide on `set -x`: https://linux.die.net/Bash-Beginners-Guide/sect_02_03.html

You can also `set -E` and `trap 'echo error on line $LINENO' ERR` to just print the line number.

Bash also has a gdb-like debugger, `bashdb`.